api/xds_protocol.rst

```diff
@@ -226,20 +226,61 @@ Transport API version
+at any time when the subscribed resources change.
+
+Whenever the client receives a new response, it will send another request indicating whether or
+not the resources in the response were valid (see
+:ref:`ACK/NACK and resource type instance version <xds_ack_nack>` for details).
+
+Only the first request on a stream is guaranteed to carry the node identifier.
+The subsequent discovery requests on the same stream may carry an empty node
+identifier. This holds true regardless of the acceptance of the discovery
+responses on the same stream. The node identifier should always be identical if
+present more than once on the stream. It is sufficient to only check the first
+message for the node identifier as a result.
+
+.. _xds_ack_nack:
+
+ACK/NACK and resource type instance version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Each xDS stream begins with a
-:ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` from the client, specifying
-the list of resources to subscribe to, the type URL corresponding to the
-subscribed resources, the node identifier and an empty :ref:`version_info <envoy_api_field_DiscoveryRequest.version_info>`.
+Every xDS resource type has a version string that indicates the version for that resource type.
+Whenever one resource of that type changes, the version number is changed.
 
-In addition to resource and transport type versioning schemes above, which operate at the type
-level, Envoy has a resource instance version. Unlike the resource/transport types, this is not a
-property of the API but is instead a reflection of the specific revision of a named resource
-delivered over xDS.
+In a responses sent by the xDS server, the
+:ref:`version_info<envoy_api_field_DiscoveryResponse.version_info>` field indicates the current
+version for that resource type. The client then sends another request to the server with the
+:ref:`version_info<envoy_api_field_DiscoveryRequest.version_info>` field indicating the most
+recent valid version seen by the client. This provides a way for the server to determine when
+it sends a version that the client considers invalid.
+
+The resource type instance version is separate for each resource type. When using the aggregated
+protocol variants, each resource type has its own version even though all resource types are being
+sent on the same stream.
+
+The resource type is also separate for each xDS server (where an xDS server is identified by a
+unique :ref:`ConfigSource <envoy_api_msg_core.ConfigSource>`). When obtaining resources of a
+given type from multiple xDS servers, each xDS server will have a different notion of version.
+
+Note that the version for a resource type is not a property of an individual xDS stream but rather
+a property of the resources themselves. If the stream becomes broken and the client creates a new
+stream, the client's initial request on the new stream should indicate the most recent version
```

Comment:
/s/should/will/

Reply:
I actually intentionally left this one a bit weaker.  If the client doesn't populate this in the initial request, things will still work, so it's not a strong requirement.  It's just a little inefficient, since it means that the server will re-send resources that the client already has.